token/program-2022/src/extension/confidential_mint_burn/processor.rs

```diff
@@ -178,8 +181,8 @@ fn process_confidential_mint(
     let authority_info = next_account_info(account_info_iter)?;
     let authority_info_data_len = authority_info.data_len();
 
-    let authority = Option::<Pubkey>::from(mint_burn_extension.authority)
-        .ok_or(TokenError::NoAuthorityExists)?;
+    let authority = OptionalNonZeroPubkey::try_from(mint_authority)?;
+    let authority = Option::<Pubkey>::from(authority).ok_or(TokenError::NoAuthorityExists)?;
```

Comment:
And here, use `ok_or`:
```suggestion
    let authority = mint.base.mint_authority.ok_or(TokenError::NoAuthorityExists)?;
```